internal/heartbeat/heartbeat.go

```diff
@@ -262,16 +275,21 @@ func (s *Heartbeat) pushInformation() error {
 		Content:   content,
 		Directive: "heartbeat",
 	}
-
+	log.Debugf("pushInformation: sending content %+v; DeviceID: %s;", content, deviceId)
 	err = s.send(data)
 
-	s.pushInfoLock.Lock()
-	defer s.pushInfoLock.Unlock()
 	if err == nil {
+		s.pushInfoLock.Lock()
```

Comment:
WDYT about this:
```go
if err = s.send(data); err != nil {
   s.pushInfoLock.RLock()
   defer s.pushInfoLock.RUnlock()
   if s.firstHearbeat {
	s.data.SetPreviousHardwareInfo(nil)
   }
   log.Debugf("pushInformation: sending content results %s; DeviceID: %s;", err, deviceId)
   return err
}


s.pushInfoLock.Lock()
defer s.pushInfoLock.Unlock()
s.firstHearbeat = false
return nil
```

Reply:
Ok, I understand now what you meant by "It's more go style" => test `err` for `!=nil` and return the error instead of testing for `==nil`
I'll do that